File: cpp/binarySearch.cpp
#include <vector>
using namespace std;

int binarySearch(vector<int> array, int target) {
  // Write your code here.
	int left = 0;
	int right = array.size()-1;
	while (left <= right) {
		int middle = (left + right) / 2;
		if (array[middle] == target) {
			return middle;
		} else if (array[middle] < target) {
			left = middle + 1;
		} else {
			right = middle - 1;
		}
	}
	return -1;
}

File: python/binarySearch.py
def binarySearch(array, target):
	left = 0
	right = len(array)-1
	while left < right:
		middle = int(left + right / 2)
		if array[middle] == target:
			return middle
		elif array[middle] < target:
			left += 1
		else:
			right -= 1
	return -1

